server/bleep/src/webserver/answer.rs

```diff
@@ -949,14 +949,20 @@ impl Agent {
 
     async fn answer(&mut self, aliases: &[usize]) -> Result<()> {
         const ANSWER_ARTICLE_MODEL: &str = "gpt-4-0613";
+        const ANSWER_HEADROOM: usize = 1024; // the number of tokens reserved for the answer
 
         debug!(?aliases, "creating article response");
 
         let context = self.answer_context(aliases, ANSWER_ARTICLE_MODEL).await?;
-        let history = self.utter_history().collect::<Vec<_>>();
-
-        let system_message = prompts::answer_article_prompt(&context);
-        let messages = Some(llm_gateway::api::Message::system(&system_message))
+        let system_prompt = prompts::answer_article_prompt(&context);
+        let system_message = llm_gateway::api::Message::system(&system_prompt);
+        let history = {
+            let h = self.utter_history().collect::<Vec<_>>();
+            let system_headroom =
+                tiktoken_rs::num_tokens_from_messages("gpt-4", &[(&system_message).into()])?;
+            trim_utter_history(h, ANSWER_HEADROOM + system_headroom)?
```

Comment:
We should pass `ANSWER_ARTICLE_MODEL` into `trim_utter_history` rather than hardcoding `gpt-4`.